crates/zizmor/src/yaml_patch/mod.rs

```diff
@@ -0,0 +1,1926 @@
+            )
+        }
+
+        YamlPatchOperation::Remove { path } => {
+            if path == "/" {
+                return Err(YamlPatchError::InvalidOperation(
+                    "Cannot remove root document".to_string(),
+                ));
+            }
+            let query = parse_json_pointer_to_query(&path)?;
+            let feature = doc.query(&query)?;
+
+            // For removal, we need to remove the entire line including leading whitespace
+            let start_pos = find_line_start(content, feature.location.byte_span.0);
+            let end_pos = find_line_end(content, feature.location.byte_span.1);
+
+            let mut result = content.to_string();
+            result.replace_range(start_pos..end_pos, "");
+            Ok(result)
+        }
+    }
+}
+
+/// Convert a JSON Pointer path (like "/permissions/contents") to a yamlpath Query
+///
+/// Supports the full JSON Pointer specification (RFC 6901):
+/// - Array indices are automatically detected and parsed as integers
+/// - Special characters are unescaped (~1 becomes /, ~0 becomes ~)
+/// - Empty path components are filtered out
+///
+/// # Examples
+/// - "/permissions/contents" → Query for nested key
+/// - "/jobs/0/steps/1" → Query for array indices
+/// - "/path/with~1slash" → Query for key containing "/"
+pub fn parse_json_pointer_to_query(path: &str) -> Result<yamlpath::Query, YamlPatchError> {
+    if !path.starts_with('/') {
+        return Err(YamlPatchError::InvalidPath(format!(
+            "Path must start with '/': {}",
+            path
+        )));
+    }
+
+    let components = path[1..] // Skip the leading '/'
+        .split('/')
+        .filter(|s| !s.is_empty())
+        .map(|component| {
+            // Try to parse as integer (for array indices)
+            if let Ok(index) = component.parse::<usize>() {
+                yamlpath::Component::Index(index)
+            } else {
+                // Decode URI encoding if present
+                let decoded = component.replace("~1", "/").replace("~0", "~");
+                yamlpath::Component::Key(decoded)
+            }
+        })
+        .collect();
+
+    yamlpath::Query::new(components)
+        .ok_or_else(|| YamlPatchError::InvalidPath("Empty path".to_string()))
+}
+
+/// Serialize a serde_yaml::Value to a YAML string, handling different types appropriately
+fn serialize_yaml_value(value: &serde_yaml::Value) -> Result<String, YamlPatchError> {
+    match value {
+        serde_yaml::Value::String(s) => {
+            // For strings, we may need to quote them if they contain special characters
+            if needs_quoting(s) {
+                Ok(serde_yaml::to_string(value)?)
+            } else {
+                Ok(s.clone())
+            }
+        }
+        _ => {
+            let yaml_str = serde_yaml::to_string(value)?;
+            Ok(yaml_str.trim_end().to_string()) // Remove trailing newline
+        }
+    }
+}
+
+/// Check if a string needs to be quoted in YAML
```

Comment:
Nitpick: I'm okay with removing this entirely and having the fix mode always perform quoting.

For context: Rust/LLVM struggle to reduce the number of linear scans needed by each `||` here; see [this Godbolt link](https://godbolt.org/z/Yse3f4ff8). An alternative would be to use a regex to accomplish the same thing, but IMO that'll be brittle/hard to maintain so I'm okay with a slightly conservative quoting approach instead 🙂